custom_components/solcast_solar/__init__.py

```diff
@@ -300,48 +305,58 @@ async def handle_service_set_dampening(call: ServiceCall):
             site = call.data.get(SITE, None) # Optional site.
 
             if factors is None:
-                raise HomeAssistantError("Error processing {SERVICE_SET_DAMPENING}: No dampening factors, must be 24 comma separated float values")
-            else:
-                factors = factors.strip().replace(" ","")
-                if len(factors.split(',')) == 0:
-                    raise HomeAssistantError(f"Error processing {SERVICE_SET_DAMPENING}: Empty dampening factor, must be 24 comma separated float values")
+                raise HomeAssistantError("Error processing {SERVICE_SET_DAMPENING}: No dampening factors, must be a comma separated list of float values")
```

Comment:
Nit picking only: float factors - are they float or decimal, and what's the difference between the two.? Would it be better to describe them as numeric?

Same comment applies to the following two exceptions for the same reason.  But as noted - this is nit picking given that it is wording of exceptions 🤣

Reply:
Decimals have a fixed precision, and are used when precision is highly valuable. Floats are an approximation of a number.

The factors are stored as floats, because we don't give a rats about precision, so the exceptions are technically correct.

Using wording like "...comma separated list of numbers between zero and one" might be a bit of a mounthful, but is not a bad idea.

Reply:
The documentation elsewhere uses number.  I'm not too fussed for this, and as I said, it's just for exception text, and as you point out, alternative wording is too long.

As a database guy, I hated floats, because they were bad for business where business (rather than science) stores actual numbers, not approximations).  Store a numeric value as a float, get it back out, and wonder why it wasn't what you put in.  Lazy developers who couldn't be bothered working out the size of the numeric data would decide "lets just make it a float" and wonder why angry BJReplay was ranting again about "we're dealing with money here, guys, it's not fucking floats".

Reply:
Because work bank, I whole heartedly agree. Because solar forecast in my spare time ... not so much. 😅